include/abb_librws/rws_cfg.h

```diff
@@ -0,0 +1,263 @@
+   * \brief The upper joint bound.
+   *
+   * The unit depends on the type of joint, i.e. [rad] for rotational and [m] for linear.
+   */
+  double upper_joint_bound;
+};
+
+/**
+ * \brief Representation of a 'Joint' instance.
+ */
+struct Joint
+{
+  /**
+   * \brief The instance's name.
+   */
+  std::string name;
+
+  /**
+   * \brief The joint's logical axis.
+   */
+  int logical_axis;
+
+  /**
+   * \brief The joint's kinematic axis number.
+   */
+  int kinematic_axis_number;
+
+  /**
+   * \brief The arm used by the joint.
+   */
+  std::string use_arm;
+
+  /**
+   * \brief The transmission used by the joint.
+   */
+  std::string use_transmission;
+};
+
+/**
+ * \brief Representation of a 'Mechanical Unit' instance.
+ */
+struct MechanicalUnit
+{
+  /**
+   * \brief The instance's name.
+   */
+  std::string name;
+
+  /**
+   * \brief The robot used by the mechanical unit.
+   */
+  std::string use_robot;
+
+  /**
+   * \brief The singles used by the mechanical unit.
+   */
+  std::vector<std::string> use_singles;
+};
+
+/**
+ * \brief Representation of a 'Robot' instance.
+ */
+struct Robot
+{
+  /**
+   * \brief The instance's name.
+   */
+  std::string name;
+
+  /**
+   * \brief The robot's type.
+   */
+  std::string use_robot_type;
+
+  /**
+   * \brief The joints used by the robot.
+   */
+  std::vector<std::string> use_joints;
+
+  /**
+   * \brief The robot's base frame.
+   */
+  Pose base_frame;
+
+  /**
+   * \brief The robot or single that moves the robot's base frame.
+   */
+  std::string base_frame_moved_by;
+};
+
+/**
+ * \brief Representation of a 'Single' instance.
+ */
+struct Single
+{
+  /**
+   * \brief The instance's name.
+   */
+  std::string name;
+
+  /**
+   * \brief The single's type.
+   */
+  std::string use_single_type;
+
+  /**
+   * \brief The joint used by the single.
+   */
+  std::string use_joint;
+
+  /**
+   * \brief The single's base frame.
+   */
+  Pose base_frame;
+
+  /**
+   * \brief The robot or single that moves the single's base frame.
+   */
+  std::string base_frame_coordinated;
+};
+
+/**
+ * \brief Representation of a 'Transmission' instance.
+ */
+struct Transmission
+{
+  /**
+   * \brief The instance's name.
+   */
+  std::string name;
+
+  /**
+   * \brief Specifies if the corresponding joint uses rotational motion (otherwise linear).
+   */
+  bool rotating_move;
+};
+} // end namespace moc
+
+/**
+ * \brief Namespace for types belonging to the controller (SYS) domain.
+ */
+namespace sys
+{
+/**
+ * \brief Representation of a 'Mechanical Unit Group' instance.
+ */
+struct MechanicalUnitGroup
+{
+  /**
+   * \brief The instance's name.
+   */
+  std::string name;
+
+  /**
+   * \brief The robot (with TCP) used by the group.
+   */
+  std::string robot;
+
+  /**
+   * \brief The mechanical units (without TCP) used by the group.
+   */
+  std::vector<std::string> mechanical_units;
+};
+
+/**
+ * \brief Representation of a 'Present Option' instance.
+ */
+struct PresentOption
```

Comment:
Preset is different from *active* I gather?

Reply:
In this case, then _active_ and _present_ are the same.

My bad if my previously commenting has been confusing.